frontend/packages/dev-console/src/components/pipelines/modals/start-pipeline/StartPipelineModal.tsx

```diff
@@ -31,11 +34,28 @@ const StartPipelineModal: React.FC<StartPipelineModalProps & ModalComponentProps
       ...workspace,
       type: 'EmptyDirectory',
     })),
-    secretOpen: false,
+    newSecrets: [],
+    secretForm: secretFormDefaultValues,
   };
 
   const handleSubmit = (values: StartPipelineFormValues, actions): void => {
     actions.setSubmitting(true);
+    const createSecrets = [];
+    values.newSecrets.map((data) => {
+      const secret = k8sCreate(SecretModel, data);
+      createSecrets.push(secret);
+      return createSecrets;
+    });
+
+    Promise.all(createSecrets)
+      .then((res) => {
```

Comment:
Favour typing early instead of using `as`.

```suggestion
      .then((res: SecretKind[]) => {
```